Create ui-c8y-1023-4-1-make-sure-language-is-actually-supported-10376.md (#3947)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-4-1-make-sure-language-is-actually-supported-10376.md

---
date: ""

> title: make sure language is actually supported (#10376)

Reviewer: ```suggestion
title: "Ensure user is not ending up on a blank screen in case it's preferred language was stored in an unexpected format"
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65299
version: 1023.4.1
---

> Co-authored-by: Tristan Bastian <tristan.bastian@cumulocity.com>

Reviewer: ```suggestion
In some cases, the user preferred language was stored in an unexpected format on the user object (as an object instead of a string). The UI implementation expected a string and was not able to handle an object instead. For an object, it then failed to initialize the application and the user ended up on a blank screen.

This behavior has been adjusted and the application will initialize in these cases as expected.
```

---
date: ""

> title: "Ensure user is not ending up on a blank screen in case it's preferred language was stored in an unexpected format"

Reviewer 2: ```suggestion
title: Application initializes properly in case of an unexpected format for the preferred language
```

> title: "Ensure user is not ending up on a blank screen in case it's preferred language was stored in an unexpected format"

Reviewer 2: @reey Does "application" here mean "platform"?

Reviewer: I do not see any usage of the term `application`?

Reviewer 2: Right, because the title has been modified. But in the description text, we still mention "application" two times.

Reviewer: With application I mean the UI application.

> title: "Ensure user is not ending up on a blank screen in case it's preferred language was stored in an unexpected format"

Reviewer 3: ```suggestion
title: "Prevent blank screen when preferred language is stored in an invalid format"
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65299
version: 1023.4.1
---

> In some cases, the user's preferred language was stored in an unexpected format in the user object (as an object instead of a string). The UI implementation expected a string and couldn't handle an object. As a result, initializing of the application failed, and the user ended up on a blank screen.

Reviewer 2: ```suggestion
In some cases, the user's preferred language was stored in an unexpected format in the user object (as an object instead of a string). The UI implementation expected a string and couldn't handle an object. As a result, initializing of the UI application failed, and the user ended up on a blank screen.
```

> In some cases, the user preferred language was stored in an unexpected format on the user object (as an object instead of a string). The UI implementation expected a string and was not able to handle an object instead. For an object, it then failed to initialize the application and the user ended up on a blank screen.

Reviewer 2: ```suggestion
In some cases, the user's preferred language was stored in an unexpected format in the user object (as an object instead of a string). The UI implementation expected a string and couldn't handle an object. As a result, initializing of the application failed, and the user ended up on a blank screen.
```
